File: dotnet/dotnet_b2b_seeding/dotnet_service/Database/Models/Role.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotnetService.Database.Models
{
    /// <summary>
    /// Role model.
    /// Similar to the Python role.py implementation.
    /// </summary>
    [Table("roles")]
    public class Role : BaseModel
    {
        [Required]
        [Column("userId")]
        public Guid UserId { get; set; }
        
        [Required]
        [MaxLength(50)]
        [Column("roleName")]
        public string RoleName { get; set; } = string.Empty;
        
        // Relationships
        [ForeignKey("UserId")]
        public virtual User User { get; set; } = null!;
    }
}

File: python/python_b2b_seeding/python_service/database/models/role.py
"""
Role model.
Similar to the Node.js role.model.ts implementation.
"""
from sqlalchemy import Column, String, ForeignKey
from sqlalchemy.orm import relationship
from database.models.base import BaseModel
from database.models.types import GUID
import uuid


class Role(BaseModel):
    """Role model."""
    __tablename__ = 'roles'
    
    # Explicitly define id first to ensure it appears first in database
    id = Column(GUID(), primary_key=True, default=uuid.uuid4)
    
    user_id = Column(GUID(), ForeignKey('users.id'), name='userId', nullable=False, index=True)
    role_name = Column(String(50), name='roleName', nullable=False)
    
    # Relationships
    user = relationship("User", back_populates="roles")

